### DISCUSSION TITLE
Can this be used for uptime/heartbeat logging?

### DISCUSSION THREAD

**Discussion by jurgenhaas** [openobserve/openobserve#946] [Q&A]
We're about to replace our ELK stacks with OpenObeserve and get really excited, as it delivers on all the excellent promisses on the front page of the project. Well done, everyone!

One component from ELK is their heartbeat tool which monitors http and icmp services, reporting into Elasticsearch directly. That's a pretty nice feature and the uptime dashboard in Kibana for it works really good.

Is there a replacement for that available in the eco system here? Any best practices for doing this with other tools? Or should there be a heartbeat ingestion support similar to the filebeat ingestion?

As a workaround, we could continue using heartbeat and let it output to a file which we could then feed into OpenObserve, but maybe there's a better way?

**Comment by prabhatsharma** [comment]
> as it delivers on all the excellent promises on the front page of the project. Well done, everyone!

Thanks for the kind words @jurgenhaas  . What is your current environment? Do you operate kubernetes?

Prometheus black-box exporter is almost a one-to-one mapping in this case. It offers the same functionality as that of elastic heartbeats. 

You will need to run:
1. Blackbox exporter and configure it to monitor the endpoints/hosts you want to monitor
2. Run a prometheus server to scrape the details from the blackbox exporter and send it to OpenObserve. 

You can then create charts and send alerts based on your needs.

https://github.com/prometheus/blackbox_exporter

**Reply by jurgenhaas** [reply]
Thanks @prabhatsharma this Prometheus black-box looks great, just we don't have Prometheus in use yet. But there have been many occations where we considered it in the past, so time may have come to go for it now.

> What is your current environment? Do you operate kubernetes?

No Kubernetes at this point. We're a Drupal agency and host a number of websites on around 50 root servers of our own or for clients. This is where we collect and aggregate alls available logs from OS, services and applications to monitor, analyze and alert.

**Comment by jurgenhaas** [comment]
Now got Prometheus with Blackbox Exporter up and running and collecting data. I've now added my OpenObserve instance as `remote_write` to Prometheus and it seems to be doing something, but I'm having a couple of issues:

- This creates a stream for each metrics, is that intended?
- Each of those streams got created but has no documents. But when I go to the metrics section in OpenObserve I'm seeing some graphs. But I'm not seeing which http target was down or slow as an example. Am I missing something?
- The logs are showing some details that I'm not yet understanding:

OpenObserve shows this, every 5 seconds:

```
[2023-06-20T11:17:17Z INFO  actix_web::middleware::logger] 169.254.0.1 "POST /api/default/prometheus/api/v1/write HTTP/1.1" 200 0 "5796" "-" "Prometheus/2.44.0" 0.004246
```

While this is happening, no log output on the Prometheus side of thing. But then, once a minute, both ends show a single log line each:

OpenObserve:

```
[2023-06-20T11:16:21Z INFO  openobserve::service::compact::file_list] [COMPACT] waiting for stream: default/metrics/probe_http_uncompressed_body_length, offset: 1687255200000000
```

Prometheus:

```
ts=2023-06-20T11:16:41.352Z caller=dedupe.go:112 component=remote level=warn remote_name=80f302 url=https://logs2.example.com/api/default/prometheus/api/v1/write msg="Failed to send batch, retrying" err="Post \"https://logs2.lakedrops.com/api/default/prometheus/api/v1/write\": EOF"
```

@prabhatsharma can you help me with that one or point me somewhere to ask these questions?

**Comment by prabhatsharma** [comment]
Let me create an article explaining how to do this best. Am traveling and in flux, so give me this week.

**Reply by jurgenhaas** [reply]
Any news on this?

**Reply by prabhatsharma** [reply]
Hey bud, I created a video explaining this - https://www.youtube.com/watch?v=wKHJdwYSXjA . Hope this helps. Do let us know.

**Reply by jurgenhaas** [reply]
This is awesome, thanks @prabhatsharma for the video. I'm up and running now.

In order to replace the current heartbeat application, I'm looking for metrics panels like this:

- a table with each target and its latest status
- a number counting all the targets being up and another number counting all the targets being down

Is that possible?

**Reply by jurgenhaas** [reply]
Ah, looks like I can do that by shortening the time range for that dashboard and then using filters.

**Reply by prabhatsharma** [reply]
Yeah, filters would be appropriate for this. You could also use custom SQL and use data from that for your charts.
